src/test/java/org/folio/rest/api/RequestBatchAPITest.java

```diff
@@ -0,0 +1,192 @@
+import static org.folio.rest.api.StorageTestSuite.storageUrl;
+import static org.folio.rest.support.ResponseHandler.empty;
+import static org.folio.rest.support.ResponseHandler.json;
+import static org.folio.rest.support.builders.RequestRequestBuilder.OPEN_AWAITING_PICKUP;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.Is.is;
+
+import java.net.URL;
+import java.util.Arrays;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.folio.rest.support.ApiTests;
+import org.folio.rest.support.JsonResponse;
+import org.folio.rest.support.Response;
+import org.folio.rest.support.builders.RequestRequestBuilder;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+
+public class RequestBatchAPITest extends ApiTests {
+
+  @Before
+  public void beforeEach() throws Exception {
+    StorageTestSuite.deleteAll(requestStorageUrl());
+  }
+
+  @After
+  public void checkIdsAfterEach() {
+    StorageTestSuite.checkForMismatchedIDs("request");
+  }
+
+  @Test
+  public void canUpdateRequestPositionsInBatch() throws Exception {
+    UUID itemId = UUID.randomUUID();
+
+    // 1st create some sample requests for an item
+    JsonObject firstRequest = createRequestForItemAtPosition(itemId, 1);
+    JsonObject secondRequest = createRequestForItemAtPosition(itemId, 2);
+
+    // 2nd: Swap positions for these requests
+    JsonObject firstRequestWithNoPosition = firstRequest.copy();
+    JsonObject secondRequestWithNoPosition = secondRequest.copy();
+
+    // 2.1 - release previous positions
+    secondRequestWithNoPosition.remove("position");
+    firstRequestWithNoPosition.remove("position");
+
+    // 2.2 - set new positions
+    firstRequest.put("position", 2);
+    secondRequest.put("position", 1);
+
+    CompletableFuture<Response> putCompleted = new CompletableFuture<>();
+    JsonObject allRequests = buildBatchUpdateRequest(firstRequestWithNoPosition,
+      secondRequestWithNoPosition, firstRequest, secondRequest);
+
+    client.put(batchRequestStorageUrl(), allRequests, TENANT_ID, empty(putCompleted));
+
+    Response response = putCompleted.get(5, TimeUnit.SECONDS);
+    assertThat(response.getStatusCode(), is(204));
+
+    // 3rd: Retrieve requests from DB and assert changes.
+    JsonObject requestsFromDb = getAllRequestsForItem(itemId);
+
+    JsonObject firstRequestFromDb = requestsFromDb.getJsonArray("requests").getJsonObject(0);
+    JsonObject secondRequestFromDb = requestsFromDb.getJsonArray("requests").getJsonObject(1);
+
+    assertThat(requestsFromDb.getInteger("totalRecords"), is(2));
+    assertThat(firstRequestFromDb, is(firstRequest));
+    assertThat(secondRequestFromDb, is(secondRequest));
+  }
+
+  @Test
+  public void canRollbackTransactionOnSamePositionConstraintViolation() throws Exception {
```

Comment:
```suggestion
  public void willAbortBatchUpdateForRequestsAtTheSamePositionInAnItemsQueue() throws Exception {
```

Reply:
I think it could be useful to have examples/tests where the requests that conflict are not both in the batch update.

For example, introduce requestA and requestB in positions 1 and 2 respectively. The update requestB to position 2 in a batch and confirm that the action is not performed.

And a test for when there are requests for different items in the same batch, and they don't cause the operation to be aborted.

Reply:
I've added these test cases.